app/workers/fast_track/disability_compensation_job.rb

```diff
@@ -52,16 +51,47 @@ def perform(form526_submission_id, full_name)
       end
     end
 
+    # class AccountNotFoundError < RuntimeError; end
+
     private
 
-    def bp_readings(client)
-      @bp_readings ||= client.list_resource('observations')
-      @bp_readings.present? ? FastTrack::HypertensionObservationData.new(@bp_readings).transform : []
-    end
+    class LighthouseVetsHealth
+      def initialize(form526_submission)
+        @form526_submission = form526_submission
+      end
+
+      def client
+        @client ||= Lighthouse::VeteransHealth::Client.new(account.icn.presence)
+      end
+
+      def account
+        user_uuid = @form526_submission.user_uuid.presence
+        edipi = @form526_submission.auth_headers['va_eauth_dodedipnid'].presence
+        account = Account.find_by(idme_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(logingov_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(edipi: edipi) if edipi
+        # Should we do this instead of expecting a NoMethodError?
+        # raise AccountNotFoundError.new("for user_uuid: #{user_uuid}") unless account&.icn
+        account
+      end
+
+      def bp_readings
+        @bp_readings ||= client.list_resource('observations')
+      end
 
-    def medications(client)
-      @medications ||= client.list_resource('medications')
-      @medications.present? ? FastTrack::HypertensionMedicationRequestData.new(@medications).transform : []
+      def observation_data
+        @observation_data ||=
+          bp_readings.present? ? FastTrack::HypertensionObservationData.new(bp_readings).transform : []
+      end
+
+      def medications
+        @medications ||= client.list_resource('medications')
+      end
+
+      def medication_request_data
+        @medication_request_data ||=
+          medications.present? ? FastTrack::HypertensionMedicationRequestData.new(medications).transform : []
```

Comment:
Same here - I don't see any reason to memoize this.

Reply:
Currently `medication_request_data` is called once, but there's nothing preventing us from calling it multiple times. I can remove the memoization for this and `observation_data` if it feels complex.

Reply:
I hear you but these data parsing classes `FastTrack::HypertensionMedicationRequestData` and `FastTrack::HypertensionObservationData` are idempotent, not very memory intensive, and aren't calling external services. There's actually no harm in them getting called again if they were to be called again. I think memoizing here is just adding unnecessary complexity.